build/build.py

```diff
@@ -969,9 +969,10 @@ def configureCompilerOptions(self):
         # specifications properly.
         # Skipping warning 4512 about being unable to generate an assignment
         # operator (since we often do this intentionally).
+	# warning C4100: '...' : unreferenced formal parameter
         # For warnings, use /W4 because /Wall
         # gives us tons of warnings in the VS headers themselves
-        warningFlags = '/W4 /wd4290 /wd4512'
+        warningFlags = '/W4 /wd4290 /wd4512 /wd4100'
```

Comment:
The only minor hesitation I have is that if we have code in a header file which has unreferenced formal parameters (only something inlined or templated), and we don't remove the unreferenced parameter names because we don't see this warning, and then someone else is using our stuff in their own build system where they don't put in this warning flag, they will suffer through extra warnings.  That being said, I think it's worth the tradeoff vs. practically speaking going through all the code and updating for this, so I think this is good.